src/box/sql.c

```diff
@@ -76,15 +79,24 @@ sql_expr_delete_cb(struct Expr *expr)
 	sql_expr_delete(expr);
 }
 
+uint32_t
+sql_flags(void)
+{
+	if (default_sql_flags == 0)
+		return 0;
```

Comment:
What is this check for? Why do you need this variable, anyway? Can't you use `SQL_DEFAULT_FLAGS` here instead?

Also, this function had better be called `sql_default_flags()`, because it returns the *default* value of session flags, not just *all* flags.

Reply:
Prior to this patch, all sessions created before `box.cfg{}` had `sql_flags` == 0. Changing them to something else would be a behavior change. Generally speaking, I should add another boolean variable to indicate that `default_sql_flags` has been initialized, but I think we expect default initialized flags to not be 0.

Reply:
> Prior to this patch, all sessions created before `box.cfg{}` had `sql_flags` == 0.

Why? Looks like a bug... Only local sessions can be created before `box.cfg` so I think it'd be fine to fix this along the way.

```
tarantool> box.session.settings
---
- sql_full_metadata: false
  sql_reverse_unordered_selects: false
  sql_full_column_names: false
  sql_vdbe_debug: false
  sql_select_debug: false
  sql_parser_debug: false
  sql_default_engine: memtx
  sql_seq_scan: false
  sql_recursive_triggers: false
...

tarantool> box.cfg{log_level = 'warn'}
---
...

tarantool> box.session.settings
---
- sql_full_metadata: false
  sql_reverse_unordered_selects: false
  sql_full_column_names: false
  sql_vdbe_debug: false
  sql_select_debug: false
  sql_parser_debug: false
  sql_default_engine: memtx
  sql_seq_scan: true
  sql_recursive_triggers: true
...
```

Reply:
SQL can't be used before `box.cfg` so I don't think we should preserve this behavior.

Reply:
Fixed.